code/modules/hydroponics/hydroponics.dm

```diff
@@ -941,8 +941,8 @@
 /obj/machinery/hydroponics/soil/update_status_light_overlays()
 	return // Has no lights
 
-/obj/machinery/hydroponics/soil/attackby(obj/item/O, mob/user, params)
-	if(O.tool_behaviour == TOOL_SHOVEL && !istype(O, /obj/item/shovel/spade)) //Doesn't include spades because of uprooting plants
+/obj/machinery/hydroponics/soil/attackby_secondary(obj/item/O, mob/user, params)
+	if(O.tool_behaviour == TOOL_SHOVEL) //Spades can still uproot plants on left click
```

Comment:
- You'll also want to `return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN` after deleting the soil
- Maybe put this after a do_after so people don't accidentally delete their plants

Reply:
Hard agree on that second point, Even just a 1 second do_after would be a nice addition.

Reply:
Alright, give me a few moments for my PC to start up and I'll put those two in.